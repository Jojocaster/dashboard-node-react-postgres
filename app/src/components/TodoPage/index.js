import { connect } from 'react-redux';

import { fetchTasks, removeTask , toggleTask } from 'actions/tasks';
import { fetchTodos } from 'actions/todos';

import TodoPage from './todoPage';

const getFilteredTasks = (tasks, {params}) => {
  if(params.id) {
    return tasks.filter(task => {
      return task.todoId === parseInt(params.id, 10);
    });
  } else {
    return tasks;
  }
}

const mapStateToProps = (state, ownProps) => {
  return {
    todos: state.todos,
    tasks: getFilteredTasks(state.tasks.items, ownProps.match)
  }
}

const mapDispatchToProps = dispatch => {
  return {
    fetchTasks: () => {
      dispatch(fetchTasks())
    },
    removeTask: task => {
      dispatch(removeTask(task))
    },
    toggleTask: task => {
      dispatch(toggleTask(task))
    },
    fetchTodos: () => {
      dispatch(fetchTodos());
    }
  }
}

const TodosContainer = connect(mapStateToProps, mapDispatchToProps)(TodoPage);

export default TodosContainer;