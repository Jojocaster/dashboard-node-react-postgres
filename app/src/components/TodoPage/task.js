import React from 'react';
import Moment from 'react-moment';
import classNames from 'classnames/bind';
import AnimateHeight from 'react-animate-height';

import styles from './task.scss';
const cx = classNames.bind(styles);

class Task extends React.Component {
  constructor(props) {
    super(props);
    this.state = {open: false};

    this.toggleState = this.toggleState.bind(this);
  }

  toggleState() {
    const {task} = this.props;
    if(!task.content) return;

    this.setState({
      open: !this.state.open
    });
  }

  render() {
    const {task, removeTask, toggleTask, i} = this.props;
    const {open} = this.state;

    const taskClassName = cx({task: true, active: task.complete});
    const contentClassName = cx({content: true, contentOpen: open});

    return(
      <span onClick={this.toggleState} className={taskClassName}>
        <span className={styles.todo}>{task.todoId}</span>
        <span className={styles.title}>{task.title}</span>
        <span className={styles.date}>
          <Moment className={styles.day} format="DD">{task.due}</Moment>
          <Moment className={styles.month} format="MMM 'YY">{task.due}</Moment>
        </span>
        <AnimateHeight contentClassName={styles.contentWrapper} duration={200} height={open ? "auto" : 0}>
          <div className={contentClassName}>{task.content}</div>
        </AnimateHeight>
      </span>
    )
  }
}

// const Task = ({task, removeTask, toggleTask, i}) => {


//   return (
//     <span onClick={toggleTask} className={className}>
//       <span className={styles.todo}>{task.todoId}</span>
//       <span className={styles.title}>{task.title}</span>
//       <span className={styles.date}>
//         <Moment className={styles.day} format="DD">{task.due}</Moment>
//         <Moment className={styles.month} format="MMM 'YY">{task.due}</Moment>
//       </span>
//     </span>
//   )
  // <Card className={task.complete ? 'task active' : 'task'}>
  //   <CardContent>
  //     <Typography type="body1">{task.title}</Typography>
  //     <IconButton onClick={removeTask} className={classes.button} aria-label="Delete">
  //       <DeleteIcon />
  //     </IconButton>
  //     <IconButton onClick={toggleTask} className={classes.button} aria-label="Delete">
  //       <DoneIcon />
  //     </IconButton>
  //   </CardContent>
  // </Card>
  // <div onClick={onTaskClick}>{task.content}</div>
// }

export default Task;