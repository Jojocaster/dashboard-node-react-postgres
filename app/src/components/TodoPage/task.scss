@import '../../styles/variables';

$todoWidth: 100px;
$dateWidth: 150px;

.task {
  background-color: white;
  border-radius: 2px;
  display: block;
  margin: 10px 0;
  padding: 20px 20px;
  position: relative;
  text-align: left;
  width: 100%;
  &:after {
    background-color: #2ecc71;
    content: '';
    height: 100%;
    opacity: 0;
    position: absolute;
    right: 0;
    top: 0;
    transform: scaleY(0);
    transition: transform .2s .1s $materialEase, opacity .2s $materialEase;
    width: 3px;
  }
  &.active:after{
    opacity: 1;
    transform: scaleY(1);
  }
  span {
    display: inline-block;
    padding: 0 10px;
    vertical-align: middle;
  }
  .todo {
    color: $appColor;
    font-weight: 700;
    text-align: center;
    width: $todoWidth;
  }
  .title {
    width: calc(100% - #{$todoWidth} - #{$dateWidth});
  }
  .date {
    text-align: right;
    width: $dateWidth;
    .day {
      font-size: 30px;
      font-weight: 700;
    }
    .month {
      color: $fontGray;
      font-size: 12px;
      font-weight: 700;
      margin-left: 5px;
      text-transform: uppercase;
    }
  }
}