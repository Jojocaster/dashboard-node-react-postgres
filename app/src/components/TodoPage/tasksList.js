import React, {Component} from 'react';
import PropTypes from 'prop-types';

import Task from './task';
// import ErrorMessage from '../ErrorMessage/';

class TaskList extends Component {
  componentDidMount() {
    this.props.fetchTasks();
  }

  render() {
    const {
      tasks,
      removeTask,
      toggleTask,
      error
    } = this.props;

    return (
      <div>
        {/* <ErrorMessage error={error}/> */}
        {tasks.map(task => (
          <Task key={task.id} task={task} removeTask={() => removeTask(task)} toggleTask={() => toggleTask(task)}/>
        ))}
      </div>
    )
  }
}

TaskList.propTypes = {
  tasks: PropTypes.arrayOf(
    PropTypes.shape({
      id: PropTypes.number.isRequired,
      content: PropTypes.string.isRequired
    }).isRequired
  ).isRequired,
  removeTask: PropTypes.func.isRequired,
  toggleTask: PropTypes.func.isRequired,
  error: PropTypes.string
}

export default TaskList;