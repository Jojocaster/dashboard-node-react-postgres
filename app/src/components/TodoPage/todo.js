import React from 'react';
import PropTypes from 'prop-types';
import { NavLink } from 'react-router-dom';

const styles = {
  todo: {
    fontSize: '12px',
    fontWeight: '600',
    marginBottom: '10px',
    textTransform: 'uppercase'
  }
};

const Todo = ({todo, onTodoClick}) => (
  <NavLink to={`/todos/${todo.id}`} key={todo.id}>
    <div style={styles.todo} onClick={onTodoClick}>
      {todo.title}
    </div>
  </NavLink>
);

Todo.propTypes = {
  todo: PropTypes.object.isRequired,
  onTodoClick: PropTypes.func
}

export default Todo;