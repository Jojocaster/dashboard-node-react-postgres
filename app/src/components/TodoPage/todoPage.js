import React from 'react';

import styles from './todoPage.scss';

import Header from 'components/Header/';
import TodosList from './todoList';
import TasksList from './tasksList';


const TodoPage = (props) => (
  <section>
    <Header title="Tasks"></Header>
    <div className={styles.content}>
      <div className={styles.todos}>
        <TodosList {...props}/>
      </div>
      <div className={styles.tasks}>
        {/* <AddTask /> */}
        <TasksList {...props}/>
      </div>
    </div>
  </section>
)

export default TodoPage;