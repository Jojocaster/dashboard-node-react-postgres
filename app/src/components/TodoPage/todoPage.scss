.content {
  composes: appContainer from '../../styles/variables.scss';
  display: flex;
  flex-direction: row;
}

.todos {
  font-size: 14px;
  font-weight: 600;
  flex-basis: 200px;
  max-width: 200px;
  margin-bottom: 10px;
  padding: 20px 0;
  text-align: left;
  text-transform: uppercase;
}

.tasks {
  width: 100%;
}