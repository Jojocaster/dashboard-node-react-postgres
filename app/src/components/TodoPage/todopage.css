.todo-page {
  display: flex;
  flex-direction: row;
}

.todos-column{
  flex-basis: 200px;
  padding: 20px;
  text-align: left;
  max-width: 200px;
}

.tasks-column {
  width: 100%;
}