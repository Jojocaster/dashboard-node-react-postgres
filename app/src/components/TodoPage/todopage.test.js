import React from 'react';
import {shallow, mount} from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import TodoPage from './';
import { Router } from 'react-router';

describe('ToDoPage', () => {
  
  it('renders without crashing', () => {
    const wrapper = shallow(<TodoPage />); 
    expect(shallowToJson(wrapper.children)).toMatchSnapshot();
  });

  it('should have props', () => {
    const wrapper = mount(<Router ><TodoPage /></Router>); 
    console.log(wrapper.props());
  })
})
  