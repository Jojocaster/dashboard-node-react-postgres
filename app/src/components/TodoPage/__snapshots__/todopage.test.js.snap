// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ToDoPage renders without crashing 1`] = `null`;
